Cargo.toml

```diff
@@ -16,7 +16,7 @@ path = "src/api/schema/gen.rs"
 required-features = ["default-no-api-client"]
 
 [profile.bench]
-debug = true
+debug = false
```

Comment:
Why did we do this? This has to have a comment justifying why we're benching with full debuginfo.

Reply:
Looks like it was added in https://github.com/timberio/vector/pull/488 to enable profiling of benchmarks. I agree that it shouldn't be enabled by default though, it can conditionally be enabled as-needed.

Reply:
I ended up re-enabling it because I needed to profile my benches. So it kind of makes sense to keep since in the very PR I had questions about it I ended up using it. So I've removed the removal of it from this PR as well.
Some comments would be useful though.

If you'd like I can re-add the removal or add comments.

Reply:
:+1: it doesn't seem like it should affect performance, but I'll try it both ways to see. Assuming it doesn't, I'm ok with leaving it on with a comment.

Reply:
It might affect the performance, and it will most definitely affect the binary size. However, logically, our benches must follow whatever settings we have in release builds. If we don't have the debuginfo in release - we can disable it in benches.

It would be really interesting to know whether it affects performance. If you're going to experiment with this either way - please add a commit here or create a new PR as you see fit based on your findings.

Reply:
I tried running a specific bench a couple of times with it on and off:

```
cargo bench --no-default-features --features "benches remap-benches" --bench remap downcase
```

There didn't appear to be any difference, but I agree with you that it'd be better if the benchmark profile more closely matched the release profile so let's remove it; or leave it in there, but commented out, so that people can easily enable it when profiling.

Reply:
Sure, let's do it in another PR though - that way it will be easier to track the differences caused by this change.